### PR TITLE
[Enhancement] - Improve Detail View Performance

### PR DESCRIPTION
**Pull request by FuchsDominik**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server.html).
- [x] Server: I added `@PreAuthorize` and check the course groups for all new REST Calls (security)
- [x] Server: I implemented the changes with a good performance and prevented too many database calls
- [x] Server: I documented the Java code using JavaDoc style.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I added `authorities` to all new routes and check the course groups for displaying navigation elements (links, buttons)
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes

### Motivation and Context

For very big courses, fetching the data can take quite long. People who want to see basic course information need to wait until all the data is fetched.

### Description

We separate fetching the basic course information and the complex statistic data, so we can see the basic information from the beginning on.
So see that we still fetch information, we add spinning icons to each graph

### Steps for Testing

1. Log in to Artemis
2. Navigate to Course Management
3. Navigate into a Detail Page by clicking in the course header
4. Make sure that while the data is getting fetched, spinning icons are shown for each graph

### Screenshots

A screencast how it should look:

https://user-images.githubusercontent.com/44401560/117706958-3bd31280-b1ce-11eb-9c22-a3988db093a0.mov

You can already see basic course information while the graphs are loading:
Screenshot 2021-05-10 at 20 24 18

### CODE REVIEW

**Review by JohannesStoehr — APPROVED** [review]
Tested on ts1, works as expected. Code looks good as well

**Review by FrankeLukas — DISMISSED** [review]
Appears to be much more responsive now, good job.

File: src/main/webapp/app/course/manage/detail/course-detail-doughnut-chart.component.ts [hunk 1]
```diff
@@ -41,13 +42,14 @@ export class CourseDetailDoughnutChartComponent implements OnInit {
         },
     ];
 
-    ngOnInit(): void {
-        this.stats = [this.currentAbsolute, this.currentMax - this.currentAbsolute];
-        if (this.currentMax === 0) {
-            // [0, 0] will lead to the chart not being displayed - is further handled in the option tooltips
+    ngOnChanges(): void {
+        if (this.currentAbsolute == undefined && !this.receivedStats) {
             this.doughnutChartData[0].data = [-1, 0];
```
[the remark this answers is not in the document]
Comment (FuchsDominik) [reply src/main/webapp/app/course/manage/detail/course-detail-doughnut-chart.component.ts:49]: Good point, i moved the comment to the first occurrence and elaborated a bit

**Review by KiraNickel — APPROVED** [review]
Tested on ts3, everything works as expected :)

File: src/main/webapp/app/course/manage/detail/course-detail-doughnut-chart.component.html [hunk 2]
```diff
@@ -2,8 +2,11 @@
     <h4 class="text-center">{{ doughnutChartTitle }}</h4>
     <div class="doughnut-chart-container">
         <div class="doughnut-chart-text">
-            <h2 class="text-center">{{ currentPercentage }} %</h2>
-            <h4 class="text-center">{{ currentAbsolute }} / {{ currentMax }}</h4>
+            <h2 *ngIf="receivedStats" class="text-center">{{ currentPercentage }} %</h2>
```
[the remark this answers is not in the document]
Comment (FuchsDominik) [reply src/main/webapp/app/course/manage/detail/course-detail-doughnut-chart.component.html]: We actually use both in our code^^ But more cases without a white space so lets do it like this 👍

**Review by joschahenningsen — APPROVED** [review]
Tested on ts1, worked as expected. :+1:

**Review by sinangunduz — APPROVED** [review]
tested on ts1, works as expected.

**Review by canyelten — APPROVED** [review]
Tested on TS1. Works as expected.

**Review by NicolasUllmann — APPROVED** [review]
Tested on Server 1. Everything worked as expected.

**Review by katjanakosic — APPROVED** [review]
Woks as expected

**Review by real-dmy — APPROVED** [review]
Tested on TS1. Re-approve. Works as expected.

**Pull request merged**